### PR TITLE
DOC: moderize pong demo

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

Updated the pong CLI program to work on all interactive backends (gtk, wx, qt4, qt5, tk tested).

 - pong driver now works on all backends
 - fixed code-rot in `set_offsets`
 - cleaned up code style

## PR Checklist

- na Has Pytest style unit tests
- [x] Code is PEP 8 compliant 
- na New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- na Added an entry to doc/users/whats_new.rst if major new feature
- na Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/event_handling/pipong.py [hunk 1]
```diff
@@ -196,23 +212,25 @@ def draw(self, evt):
...
                 if puck.update(self.pads):
                     # we only get here if someone scored
-                    self.pads[0].disp.set_label("   " + str(self.pads[0].score))
-                    self.pads[1].disp.set_label("   " + str(self.pads[1].score))
+                    self.pads[0].disp.set_label(
+                        "   " + str(self.pads[0].score))
+                    self.pads[1].disp.set_label(
+                        "   " + str(self.pads[1].score))
                     self.ax.legend(loc='center')
                     self.leg = self.ax.get_legend()
```
Comment (QuLogic) [inline examples/event_handling/pipong.py]: Is this not what the above call returns?

File: examples/event_handling/pipong.py [hunk 2]
```diff
@@ -196,23 +212,25 @@ def draw(self, evt):
...
-                    #self.leg.draw_frame(False) #don't draw the legend border
                     self.leg.get_frame().set_alpha(.2)
```
Comment (QuLogic) [inline examples/event_handling/pipong.py]: Pass `framealpha=.2` above?

File: examples/event_handling/pipong.py [hunk 3]
```diff
@@ -196,23 +212,25 @@ def draw(self, evt):
...
-                    plt.setp(self.leg.get_texts(), fontweight='bold', fontsize='xx-large')
+                    plt.setp(self.leg.get_texts(), fontweight='bold',
+                             fontsize='xx-large')
```
Comment (QuLogic) [inline examples/event_handling/pipong.py]: Pass `fontsize` above, maybe `fontweight` too.

File: examples/event_handling/pipong.py [hunk 4]
```diff
@@ -196,23 +212,25 @@ def draw(self, evt):
...
                     self.leg.get_frame().set_facecolor('0.2')
```
Comment (QuLogic) [inline examples/event_handling/pipong.py]: Pass `facecolor` above?

**Comment by ivanov** [comment]
Bless you, @tacaswell - this was one of my first contributions to matplotlib! So many memories flooding back - I remember exactly what cafe I was at in Berkeley when I wrote this 8 and a half years ago :) 👍

File: examples/event_handling/pong_sgskip.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+A small game demo using GTK.
+
+"""
+from __future__ import print_function, division
+import time
+
+
+import matplotlib.pyplot as plt
+import pipong
```
Comment (WeatherGod) [inline examples/event_handling/pong_sgskip.py]: could this be more explicit by doing `from . import pipong`?

Comment (QuLogic) [inline examples/event_handling/pong_sgskip.py]: It's not a package, so that wouldn't work.

Comment (story645) [inline examples/event_handling/pong_sgskip.py]: Sphinx has a...download directive that can be used to  link to pipong to make it explicit that this is  a local library. Did something like that for the unit examples.

File: examples/event_handling/pong_sgskip.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
+"""
+========
+Pong GTK
+========
```
Comment (jenshnielsen) [inline examples/event_handling/pong_sgskip.py]: This is no longer GTK specific right?

**Comment by ivanov** [comment]
👍

**Pull request merged**